### PR TITLE
Added exists-maven-plugin.

### PR DESCRIPTION
**Pull request by MMaiero**
Plugins added both in manifest_pom and target-platform poms.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

### CODE REVIEW

File: kura/manifest_pom.xml [hunk 1]
```diff
@@ -119,6 +99,8 @@
         <!-- BND baseline plugin -->
         <bnd.baseline.version>3.4.0-SNAPSHOT</bnd.baseline.version>
         <bnd.baseline.continueOnError>false</bnd.baseline.continueOnError>
+
+        <maven.deploy.skip.prop>false</maven.deploy.skip.prop>
```
Comment (cdealti) [inline kura/manifest_pom.xml]: ``maven.deploy.skip.prop`` -> ``exists.artifact.already.released``

File: kura/manifest_pom.xml [hunk 2]
```diff
@@ -510,44 +491,76 @@
...
-                                            refid="fulljre.file" />
+                                        <resourcecount when="greater" count="0" refid="fulljre.file" />
                                     </condition>
                                 </fail>
                             </target>
                         </configuration>
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```
Comment (cdealti) [inline kura/manifest_pom.xml:521]: needs build and test dependency CQ (I will open an issue)

File: kura/manifest_pom.xml [hunk 3]
```diff
@@ -510,44 +491,76 @@
...
+                        </goals>
+                        <configuration>
+                            <skipIfSnapshot>true</skipIfSnapshot>
+                            <property>maven.deploy.skip.prop</property>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (cdealti) [inline kura/manifest_pom.xml]: CQ build and test dependency needed

File: kura/manifest_pom.xml [hunk 4]
```diff
@@ -510,44 +491,76 @@
...
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <version>1.4.1</version>
+                <executions>
+                    <execution>
+                        <id>enforce-property</id>
```
Comment (cdealti) [inline kura/manifest_pom.xml]: ``enforce-property`` -> ``fail-on-artifact-already-released``

File: kura/manifest_pom.xml [hunk 5]
```diff
@@ -510,44 +491,76 @@
...
+                        <phase>verify</phase>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <requireProperty>
+                                    <property>maven.deploy.skip.prop</property>
+                                    <message>Property must be false during execution.</message>
```
Comment (cdealti) [inline kura/manifest_pom.xml]: Artifact already exists on the release repository. Forgot to bump version?

File: target-platform/pom.xml [hunk 6]
```diff
@@ -2,91 +2,103 @@
...
-				<artifactId>maven-deploy-plugin</artifactId>
-				<version>2.8.1</version>
-				<configuration>
-				</configuration>
-			</plugin>
-			<plugin>
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+        <maven.deploy.skip.prop>false</maven.deploy.skip.prop>
```
Comment (cdealti) [inline target-platform/pom.xml]: same as above

**Review by cdealti — CHANGES_REQUESTED** [review]

**Comment by MMaiero** [comment]
Absolutely, the plugins need to be added to the build and test dependency doc before merging the PR. At least we don't miss it.

**Comment by cdealti** [comment]
Please rebase & update the PR as the build has been fixed by #1441

**Comment by cdealti** [comment]
@MMaiero failing the build should be avalable out of box with version 0.0.3:

https://github.com/chonton/exists-maven-plugin/issues/1

**Comment by MMaiero** [comment]
Plugin updated to Version 0.0.3.
The build fails if an artifact already exists remotely

**Comment by MMaiero** [comment]
Rebased branch.
Created issue for the Build and test dependency.
@pierantoniomerlino I think that if it is ok for you, we can merge.

**Review by pierantoniomerlino — APPROVED** [review]

**Pull request merged**